### ISSUE TITLE
How to use plugins in callback [ios]

### ISSUE THREAD

**Issue by ozz-rjq** [rekab-app/background_locator#82]
Hello, can you please add the documentation on how to use the third-party plugins inside the callback function for IOS?

**Comment by mehdok** [comment]
Sadly I don't know a way to do that, it fix a lot of problem if we could find a way to do that.

**Comment by dacianf** [comment]
@ozz-rjq Do you mean to use for example local notification plugin in the callback? Or a plugin for making server requests?

**Comment by ozz-rjq** [comment]
@dacianf yes, plugins like local notifications or audioplayers

**Comment by dacianf** [comment]
@ozz-rjq Have you tried to do it? What errors gave you? I'm asking these because in my current project I use two plugins in the callback function, local notification and shared preferences and they work perfect.

**Comment by SandeMK** [comment]
@dacianf How did you register your plugins on AppDelegate?, because I also want to use shared preferences but the app crashes each time I run the service.

**Comment by dacianf** [comment]
@SandeMK what errors have it shown? Did you gave permissions to your ios app to run in background (inside signing & capabilities tab in xcode)? Can you post here the errors?
![image](https://user-images.githubusercontent.com/19537170/85258005-6362a480-b46f-11ea-96eb-36fa677cc4d2.png)

**Comment by SandeMK** [comment]
@dacianf yes I did give permissions just as shown on that pic, I think what caused the crashing was the plugin registrant callback method. I changed it to 

`if (!registry.hasPlugin("BackgroundLocatorPlugin")) {
        GeneratedPluginRegistrant.register(with: registry)
   }'

The app doesn't crush now, but I get location update only once, and that's when I'm starting the service. Is this the case in IOS when you are stationary?(I'm debugging on a simulator), does IOS update only when the device is moving?
because with the same code, Android does update.

**Comment by mehdok** [comment]
I just updated the wiki page for using other plugins in iOS. Please check it out [Wiki](https://github.com/rekab-app/background_locator/wiki/Use-other-plugins-in-callback#ios)

**Comment by elouiestudent** [comment]
I'm having a similar problem. I'm using this plugin and a geofencing plugin found [here](https://github.com/bkonyi/FlutterGeofencing). In the location callback function I create geofences based on the user's location. This works fine on Android, but on iOS calling any methods from the geofencing plugin in the location callback has no effect. 

Below is my AppDelegate.swift. Any help would be much appreciated!
```

func registerPlugins(registry: FlutterPluginRegistry) -> () {
  if (!registry.hasPlugin("BackgroundLocatorPlugin")) {
    BackgroundLocatorPlugin.register(with: registry.registrar(forPlugin: "BackgroundLocatorPlugin"))
  } 
  if (!registry.hasPlugin("GeofencingPlugin")) {
    GeofencingPlugin.register(with: registry.registrar(forPlugin: "GeofencingPlugin"))
  }
  if (!registry.hasPlugin("FlutterLocalNotificationsPlugin")) {
    FlutterLocalNotificationsPlugin.register(with: registry.registrar(forPlugin: "FlutterLocalNotificationsPlugin"))
  }
  if (!registry.hasPlugin("FLTSharedPreferencesPlugin")) {
    FLTSharedPreferencesPlugin.register(with: registry.registrar(forPlugin: "FLTSharedPreferencesPlugin"))
  }
  if (!registry.hasPlugin("FlutterSecureStoragePlugin")) {
    FlutterSecureStoragePlugin.register(with: registry.registrar(forPlugin: "FlutterSecureStoragePlugin"))
  }
}

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GMSServices.provideAPIKey("$GOOGLE_MAPS_API_KEY")

    GeneratedPluginRegistrant.register(with: self)

    GeofencingPlugin.setPluginRegistrantCallback(registerPlugins)
    BackgroundLocatorPlugin.setPluginRegistrantCallback(registerPlugins)

    // iOS notification setup
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }
    // Prevent "old" notifications showing up after reinstallation
    if(!UserDefaults.standard.bool(forKey: "Notification")) {
      UIApplication.shared.cancelAllLocalNotifications()
      UserDefaults.standard.set(true, forKey: "Notification")
    }
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}

**Comment by yu3fms** [comment]
> @ozz-rjq Have you tried to do it? What errors gave you? I'm asking these because in my current project I use two plugins in the callback function, local notification and shared preferences and they work perfect.

Can You please provide code on how You actually use "flutter_local_notifications" plugin in callback?

I am having problems with this plugin in callback (in the "callback" function inside "locations_services_repository.dart"), and only in iOS. It never gets executed and no error is thrown. Otherwise this plugin is working normally anywhere in the app and on Android it is working also in callback.

Also, for example, "flutter_ringtone_player" works for me in callback without registering it in AppDelegate.swift

If I try and register plugin as shown in an example in wiki page, then following error is thrown.
error: cannot find 'FlutterLocalNotificationsPlugin' in scope
           FlutterLocalNotificationsPlugin.register(with: registrar(forPlugin: "com.dexterous.flutterlocalnotifications"))

**Comment by yu3fms** [comment]
@dacianf Got it working.

There is a difference in code on wiki page https://github.com/rekab-app/background_locator/wiki/Use-other-plugins-in-callback and in the AppDelegate.swift code on line 30: https://github.com/rekab-app/background_locator/blob/239a850832766be20036552e81ccd956b1afbbaa/example/ios/Runner/AppDelegate.swift#L30

Adding "as! FlutterPluginRegistrar" resolved my issue and I can use "flutter_local_notifications" plugin both in background mode and when app is terminated.

I guess this should be fixed in documentation on wiki page so it matches the code in example.
